server/s3_motr_rw_common.cc

```diff
@@ -89,7 +98,32 @@ void motr_op_done_on_main_thread(evutil_socket_t, short events,
              S3_IEM_MOTR_CONN_FAIL_JSON);
     }
     context->on_failed_handler()();  // Invoke the handler.
+
+    if (error_code == -ETIMEDOUT) {
+      S3Option *optinst = S3Option::get_instance();
+      unsigned err_thr = optinst->get_motr_etimedout_max_threshold();
+      unsigned err_wnd = optinst->get_motr_etimedout_window_sec();
+      int64_t curtime = time(NULL);
+      if (curtime - gs_timeout_window_start > (int64_t)err_wnd) {
+        s3_log(S3_LOG_DEBUG, request_id,
+               "Reset motr ETIMEDOUT window; cur window sec %" PRId64 ";",
+               curtime - gs_timeout_window_start);
+        gs_timeout_window_start = curtime;
+        gs_timeout_cnt = 0;
+      }
+      gs_timeout_cnt++;
+      if (gs_timeout_cnt >= err_thr) {
+        s3_log(
+            S3_LOG_ERROR, request_id,
+            "Shutdown. Motr ETIMEDOUT error cnt %u reached threshold value %u "
+            "for %" PRId64 " sec with allowed window %u sec",
+            gs_timeout_cnt, err_thr, curtime - gs_timeout_window_start,
+            err_wnd);
```

Comment:
you need to ensure that for some other request we dont call s3_kickoff_graceful_shutdown() again, so either reset gs_timeout_cnt and set err_thr to a very high value here or by some other way

Reply:
done